src/validation.cpp

```diff
@@ -1074,9 +1182,76 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptMultipleTransactions(const std::
         }
     }
 
+    if (args.m_test_accept) return PackageMempoolAcceptResult(package_state, std::move(results));
+
+    if (!FinalizePackage(args, workspaces, package_state, results)) {
+        package_state.Invalid(PackageValidationResult::PCKG_TX, "submission failed");
+        return PackageMempoolAcceptResult(package_state, std::move(results));
+    }
+
     return PackageMempoolAcceptResult(package_state, std::move(results));
 }
 
+PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package, ATMPArgs& args)
+{
+    AssertLockHeld(cs_main);
+    PackageValidationState package_state;
+
+    // Check that the package is well-formed. If it isn't, we won't try to validate any of the
+    // transactions and thus won't return any MempoolAcceptResults, just a package-wide error.
+
+    // Context-free package checks.
+    if (!CheckPackage(package, package_state)) return PackageMempoolAcceptResult(package_state, {});
+
+    // All transactions in the package must be a parent of the last transaction. This is just an
+    // opportunity for us to fail fast on a context-free check without taking the mempool lock.
+    if (!IsChildWithParents(package)) {
+        package_state.Invalid(PackageValidationResult::PCKG_POLICY, "package-not-child-with-parents");
+        return PackageMempoolAcceptResult(package_state, {});
+    }
+
+    LOCK(m_pool.cs);
+    const auto& child = package[package.size() - 1];
+    // The child is never allowed to conflict with the mempool.
+    if (m_pool.exists(GenTxid::Txid(child->GetHash()))) {
+        package_state.Invalid(PackageValidationResult::PCKG_POLICY, "conflicts-with-mempool");
+        return PackageMempoolAcceptResult(package_state, {});
+    }
+    // The package must be 1 child with all of its unconfirmed parents.
```

Comment:
nit: I found the similar comment in `packages.cpp` clearer: "The package is expected to be sorted, so the last transaction is the child."

I would clarify it to:

```suggestion
    // The package is expected to be sorted, so all transactions except the last one are the unconfirmed parents
```